includes/functions.inc.php

```diff
@@ -40,39 +41,53 @@ function gdrcd_connect()
- * @param string $mode : La modalità con cui eseguire la query. Default "query"
+ * Esegue una query SQL o gestisce un risultato esistente.
+ *
+ * Questa funzione serve come un'interfaccia di astrazione per le operazioni sul database.
+ * Mantiene la retrocompatibilità con i risultati di mysqli_query e aggiunge il supporto
+ * per la gestione dei risultati restituiti da gdrcd_stmt.
+ *
+ * @param mysqli_result|array|string $sql La query SQL (stringa) o il risultato (mysqli_result o gdrcd_stmt) da gestire.
+ * @param string $mode La modalità di operazione.
  * Modalità accettate:
- *  query: esegue la query e ritorna come risultato la prima riga del resultset
- *  result: esegue la query e ritorna la risorsa MySql associata al risultato
- *  num_rows: accetta come parametro una risorsa mysqli e ritorna il numero di righe nel resultset
- *  fetch: accetta come parametro una risorsa mysqli e ritorna il successivo risultato dal resultset come array
- *  object: uguale a fetch, eccetto che ritorna un oggetto al posto di un array
- *  free: libera la memoria occupata dalla risorsa mysqli passata in $sql
- *  last_id: ritorna l'id del record generato dall'ultima query, se non era una INSERT o UPDATE ritorna 0. In questo caso $sql non viene considerato
- *  affected: ritorna il numero di record toccati dall'ultima query (INSERT, UPDATE, DELETE o SELECT). In questo caso $sql non viene considerato
+ *  - query: esegue la query e ritorna come risultato la prima riga del resultset
+ *  - result: esegue la query e ritorna la risorsa mysqli_result associata al risultato
+ *  - num_rows: accetta come parametro una risorsa mysqli_result o il risultato di gdrcd_stmt e ritorna il numero di righe
+ *  - fetch: accetta come parametro una risorsa mysqli_result o il risultato di gdrcd_stmt e ritorna il successivo risultato dal resultset come array
+ *  - object: uguale a fetch, eccetto che ritorna un oggetto al posto di un array
+ *  - free: libera la memoria occupata dalla risorsa mysqli_result o del risultato di gdrcd_stmt passato in $sql
+ *  - last_id: ritorna l'id del record generato dall'ultima query di INSERT
+ *  - affected: ritorna il numero di record toccati dall'ultima query (INSERT, UPDATE, DELETE o SELECT)
  * @return mixed un booleano in caso di esecuzione di query non SELECT e modalità 'query'. Altrimenti ritorna come specificato nella descrizione di $mode
- * @throws Exception
+ * @param bool $throwOnError Se true, solleva un'eccezione in caso di errore. Altrimenti, termina lo script.
+ * @return mixed Il risultato dell'operazione.
  */
 function gdrcd_query($sql, $mode = 'query', $throwOnError = false)
 {
     $db_link = gdrcd_connect();
 
+    // Rileva se l'input è già un risultato (object o array) o una query string
+    $isResultObject = is_object($sql) && ($sql instanceof mysqli_result);
+    $isStmtResult = (is_array($sql) && $sql['data'] instanceof StmtResultData) || (is_array($sql) && isset($sql['affected_rows']));
+
```

Comment:
`gdrcd_query()` continua a ritornare `-1` in modalità `affected` con una query `UPDATE`.
Il problema in questo caso sembra derivare dal fatto che qui viene controllata la chiave come `affected_rows` ancora
```suggestion
    $isStmtResult = (is_array($sql) && $sql['data'] instanceof StmtResultData) || (is_array($sql) && isset($sql['affected']));
```